src/core/crypto.c

```diff
@@ -1813,8 +1827,9 @@ QuicCryptoProcessData(
                 &BufferCount,
                 &Buffer);
 
-        UNREFERENCED_PARAMETER(DataAvailable);
-        CXPLAT_TEL_ASSERT(DataAvailable);
+        if (!DataAvailable) {
+            return Status;
+        }
```

Comment:
This is needed for when the function is called from *ValidationComplete functions, if we want to preserve the assert then it should be possible by adding another parameter to the function saying "There should be data available"